authorizerd.go

```diff
@@ -64,21 +68,28 @@ type provider struct {
 	policyEtagExpTime     string
 }
 
-// New return Providerd
-// This function will initialize the Providerd object with the options
-func New(opts ...Option) (Providerd, error) {
-	prov := &provider{
+type mode uint8
+
+const (
+	token mode = iota
+	jwt
+)
+
+// New return Authorizerd
+// This function will initialize the Authorizerd object with the options
+func New(opts ...Option) (Authorizerd, error) {
+	prov := &authorizer{
 		cache: gache.New(),
 	}
 
 	var err error
 	for _, opt := range append(defaultOptions, opts...) {
 		if err = opt(prov); err != nil {
-			return nil, errors.Wrap(err, "error creating providerd")
+			return nil, errors.Wrap(err, "error creating authorizerd")
 		}
 	}
 
-	if prov.pubkeyd, err = pubkey.NewPubkeyd(
+	if prov.pubkeyd, err = pubkey.New(
```

Comment:
```suggestion
	if prov.pubkeyd, err = pubkeyd.New(
```

Reply:
changing package name is boring,

Reply:
I think this also make sense, without changing package name.
```suggestion
	if prov.pubkeyd, err = pubkey.NewDaemon(
```